src/App.tsx

```diff
@@ -14,11 +14,131 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const numbers: number[] = Array.from(Array(10).keys());
+
+const goodLengthStart = '1';
+
+type State = {
+  goods: string[];
+  goodLength: string;
+  isVisible: boolean
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    goodLength: goodLengthStart,
+    isVisible: false,
+  };
+
+  reversedGoods = () => {
+    this.setState((state) => ({
+      goods: [...state.goods].reverse(),
+    }));
+  };
+
+  sortedGoods = () => {
+    this.setState((state) => ({
+      goods: [...state.goods].sort((a, b) => a.localeCompare(b)),
+    }));
+  };
+
+  resetGoods = () => {
+    this.setState({
+      goods: [...goodsFromServer],
+      goodLength: goodLengthStart,
+    });
+  };
+
+  sortByLengthGoods = () => {
+    this.setState((state) => ({
+      goods: [...state.goods].sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  changeLength = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const newGoods = goodsFromServer.filter(
+      good => good.length >= Number(event.target.value),
+    );
+
+    this.setState({
+      goodLength: event.target.value,
+      goods: newGoods,
+    });
+  };
+
+  showGoods = () => {
+    this.setState({ isVisible: true });
+  };
```

Comment:
Your idea with a button for hiding the list was nice. You can create a method updateVisibility that received value and update isVisible with this value. Or create toggleVisibility:

```suggestion
  toggleVisibility = () => {
    this.setState( currentState => ({ isVisible:  !currentState.isVisible }));
  };
```